conan/api/conan_api.py

```diff
@@ -102,12 +97,34 @@ class _ApiHelpers:
         def __init__(self, conan_api):
             self._conan_api = conan_api
             self.hook_manager = HookManager(HomePaths(self._conan_api.home_folder).hooks_path)
```

Comment:
Unless I'm missing something, this initialization is now being performed before the version check, which is what https://github.com/conan-io/conan/issues/18643 reported was making their setup fail